### PR TITLE
WX-1710 Move option for final output files

### PR DESCRIPTION
**Pull request by aednichols**
### Description

- Define new workflow option for workflow outputs mode, move or copy
- Enhance Centaur to allow nested directories of `.test` files, which is required for the next step to be sane
- Relocate regression tests for existing copy behavior so they actually run
- Add test for new move mode
- Created new `centaur-ci-us-east1` bucket in `broad-dsde-cromwell-dev` and used it to replace `cloud-cromwell-dev-self-cleaning-fast` bucket. It was really hard to tell whether one was successfully testing a cross-region move.

### Release Notes Confirmation

The move mode is documented but is not ready for a release note because we are not updating metadata yet.

#### `CHANGELOG.md`
 - [ ] I updated `CHANGELOG.md` in this PR
 - [x] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: centaur/src/main/resources/reference.conf [hunk 1]
```diff
@@ -56,9 +56,9 @@ centaur {
     genomics.endpoint-url = ${?CROMWELL_BUILD_PAPI_ENDPOINT_URL}
     genomics.location = "us-central1"
     batch.location = "us-central1"
-    auth = "Error: BA-6546 The environment variable CROMWELL_BUILD_PAPI_AUTH_MODE must be set/export pointing to a valid auth such as 'application-default'"
+    auth = "service-account"
```
Comment (aednichols) [inline centaur/src/main/resources/reference.conf:59]: Simplifying the process to run Centaur locally.

I appreciate the principle of these sentinel values that explode when you don't set an environment variable.

But for both `CROMWELL_BUILD_PAPI_AUTH_MODE` and `CROMWELL_BUILD_RESOURCES_DIRECTORY` there's only one value we ever set in whole repo.

File: centaur/README.md [hunk 2]
```diff
@@ -1 +1,21 @@
-For information on Cromwell's Integration Testing Suite, see the [Cromwell documentation on Centaur](https://cromwell.readthedocs.io/en/develop/developers/Centaur/).
\ No newline at end of file
+For information on Cromwell's Integration Testing Suite, see the [Cromwell documentation on Centaur](https://cromwell.readthedocs.io/en/develop/developers/Centaur/).
```
Comment (jgainerdewar) [inline centaur/README.md:1]: 🙌 Thank you for this.

File: centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsCopying.test [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  workflow: wdlResultsCopying/simpleWorkflow.wdl
+  options: wdlResultsCopying/gcp/options.json
+}
+
+metadata {
+  status: Succeeded
+}
+
+fileSystemCheck: "gcs"
+outputExpectations: {
```
Comment (THWiseman) [inline centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsCopying.test:20]: TIL about `fileSystemCheck` and `<<UUID>>`

File: centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsMoving.test [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+metadata {
+  status: Succeeded
+}
+
+# The `centaur-ci-us-east1` bucket is in a different region than the workflow runs
+fileSystemCheck: "gcs"
+outputExpectations: {
+    "gs://centaur-ci-us-east1/move_destination/simpleWorkflow/<<UUID>>/call-simpleStdoutTask/output.txt": 1
+    "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/simpleWorkflow/<<UUID>>/call-simpleStdoutTask/output.txt": 0
```
Comment (jgainerdewar) [inline centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsMoving.test:22]: I would have assumed `outputExpectations` is comparing with metadata but I infer from this that it's checking for the actual existence of the files?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/finalization/CopyWorkflowOutputsActor.scala [hunk 5]
```diff
@@ -97,6 +101,21 @@ class CopyWorkflowOutputsActor(workflowId: WorkflowId,
...
 
+  private def moveWorkflowOutputs(workflowOutputsFilePath: String): Future[Seq[Unit]] = {
+    val workflowOutputsPath = buildPath(workflowOutputsFilePath)
+    val outputFilePaths = getOutputFilePaths(workflowOutputsPath)
+
+    markDuplicates(outputFilePaths)
+
+    val moves = outputFilePaths map { case (srcPath, dstPath) =>
+      asyncIo.copyAsync(srcPath, dstPath) flatMap { _ =>
+        asyncIo.deleteAsync(srcPath)
```
Comment (jgainerdewar) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/finalization/CopyWorkflowOutputsActor.scala:112]: Do we have anything in the tests anywhere that validates that we only delete the file when the copy is successful?

Comment (aednichols) [reply engine/src/main/scala/cromwell/engine/workflow/lifecycle/finalization/CopyWorkflowOutputsActor.scala:112]: Added `gcpWdlResultsMovingFail` case.

File: centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsMoving.test (hunk 4 above)
Comment (aednichols) [reply centaur/src/main/resources/standardTestCases/resultsCopyingTestCases/gcpWdlResultsMoving.test:22]: Yep, documented [here](https://cromwell.readthedocs.io/en/stable/developers/Centaur/).

**Review by THWiseman — APPROVED** [review]

**Pull request merged**